__tests__/integration/template/transaction/template.test.ts

```diff
@@ -0,0 +1,346 @@
+      .addSetVarAction({
+        name: 'tk_balance',
+        call: { method: 'get_wallet_balance', token: '{token}' },
+      })
+      .addUtxoSelect({ fill: '{tk_balance}', token: '{token}', autoChange: false })
+      .addAuthoritySelect({ token: '{token}', authority: 'mint' })
+      .addAuthoritySelect({ token: '{token}', authority: 'melt' })
+      .addTokenOutput({ address: '{addr}', amount: 1, token: '{token}' })
+      .addCompleteAction({})
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    await waitForTxReceived(hWallet, tx.hash, null);
+    expect(tx.hash).toBeDefined();
+  });
+
+  it('should be able to add change with the change instruction', async () => {
+    const address = await hWallet.getAddressAtIndex(25);
+    const template = new TransactionTemplateBuilder()
+      .addSetVarAction({ name: 'addr', value: address })
+      .addSetVarAction({ name: 'token', value: tokenUid })
+      .addUtxoSelect({ fill: 100, token: '{token}', autoChange: false })
+      .addTokenOutput({ address: '{addr}', amount: 1, token: '{token}' })
+      .addDataOutput({ data: 'cafe', token: '{token}' })
+      .addChangeAction({})
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    await waitForTxReceived(hWallet, tx.hash, null);
+    expect(tx.hash).toBeDefined();
+  });
+
+  it('should be able to send tokens and authorities without using template variables', async () => {
+    const address = await hWallet.getAddressAtIndex(10);
+    const template = new TransactionTemplateBuilder()
+      .addUtxoSelect({ fill: 2 })
+      .addTokenOutput({ address, amount: 2 })
+      .addUtxoSelect({ fill: 3, token: tokenUid })
+      .addTokenOutput({ address, amount: 3, token: tokenUid })
+      .addAuthoritySelect({ authority: 'mint', token: tokenUid, count: 1 })
+      .addAuthorityOutput({ address, authority: 'mint', count: 1, token: tokenUid })
+      .addAuthoritySelect({ authority: 'melt', token: tokenUid, count: 2 })
+      .addAuthorityOutput({ address, authority: 'melt', count: 2, token: tokenUid })
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+
+    expect(tx.hash).toBeDefined();
+    if (!tx.hash) {
+      throw new Error('tx hash should be defined');
+    }
```

Comment:
suggestion: move to `expect`

Reply:
We already have an expect, but the expect does not ensure the typing so we also need the if statement.